components/Link.tsx

```diff
@@ -154,7 +155,12 @@ export const Link = Object.assign(
 
 export function AIM({ bold = true, paragraph, ...rest }: AIMProps) {
   const [chapter, section, subsection, ...components] = paragraph;
-  const name = useAIM(paragraph);
+
+  const name = React.useMemo(() => {
+    try {
+      return aim[chapter][section][subsection || 'name'];
+    } catch (e) {}
+  }, [chapter, section, subsection]);
```

Comment:
The `useAIM` hook was only used once, defeating the reusability purpose of hooks. I moved the hook content here.